### PR TITLE
8315651: Stop hiding AIX specific multicast socket errors via NetworkConfiguration (aix)

### PR DESCRIPTION
**Pull request by TOatGithub**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8315651](https://bugs.openjdk.org/browse/JDK-8315651): Stop hiding AIX specific multicast socket errors via NetworkConfiguration (aix) (**Sub-task** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15586/head:pull/15586` \
`$ git checkout pull/15586`

Update a local copy of the PR: \
`$ git checkout pull/15586` \
`$ git pull https://git.openjdk.org/jdk.git pull/15586/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15586`

View PR using the GUI difftool: \
`$ git pr show -t 15586`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15586.diff">https://git.openjdk.org/jdk/pull/15586.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15586#issuecomment-1708026161)

### CODE REVIEW

**Comment by TOatGithub** [comment]
Description:
test/lib/jdk/test/lib/NetworkConfiguration.java has special code to avoid running into the error to be resolved by the main task [JDK-8308807](https://bugs.openjdk.org/browse/JDK-8308807).
In order to have a clear picture and to potentially see other hidden bugs this extra code should be removed.
One example is java/net/MulticastSocket/Promiscuous.java, which currently fails because from NetworkConfiguration no ip4MulticastInterface is provided at all. If the AIX specific filter is removed we see the problem addressed in main task.

File: test/lib/jdk/test/lib/NetworkConfiguration.java [hunk 1]
```diff
@@ -183,9 +183,10 @@ private boolean supportsIp4Multicast(NetworkInterface nif) {
             // could be in both, IPv4 or IPv6 space). But both possible setsockopt
             // calls for either IPV6_MULTICAST_IF or IP_MULTICAST_IF return
             // EADDRNOTAVAIL. So we must skip such interfaces here.
-            if (Platform.isAix() && isIPv6Available() && !hasIp6Addresses(nif)) {
-                return false;
-            }
+            // to be fixed with JDK-8308807 -> comment this aviation
+            //if (Platform.isAix() && isIPv6Available() && !hasIp6Addresses(nif)) {
+            //    return false;
+            //}
```
Comment (AlanBateman) [inline test/lib/jdk/test/lib/NetworkConfiguration.java:189]: Can this entire comment block be removed?

Comment (TOatGithub) [reply test/lib/jdk/test/lib/NetworkConfiguration.java]: did so with latest commit

**Review by AlanBateman — APPROVED** [review]

File: test/lib/jdk/test/lib/NetworkConfiguration.java (hunk 1 above)
Comment (AlanBateman) [reply test/lib/jdk/test/lib/NetworkConfiguration.java]: Thanks, this removal looks okay to me although it's impossible to know if tests will now fail in some AIX configs.

**Review by TheRealMDoerr — APPROVED** [review]

**Pull request closed without merging**